### PR TITLE
add command line parameter for replicaset process delaying time

### PR DESCRIPTION
**Pull request by jianhuiz**
add command line parameter for the delays in replicaset controller
requires #31677
#29741

@quinton-hoole @deepak-vij @kshafiee @mwielgus

---

This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/31815)

### CODE REVIEW

**Comment by quinton-hoole** [comment]
I have some concerns about this, but don't have time to discuss today.  Do we need this for v1.4, or can it wait until next week?  I assume the latter.

**Comment by jianhuiz** [comment]
@quinton-hoole No, I didn't expected this to be merged. But #31677 should be.

**Comment by dims** [comment]
@jianhuiz @mwielgus Do we need this for 1.5? (I see a 1.5 milestone)

**Comment by dims** [comment]
@jianhuiz Can you please rebase the PR?

**Comment by jianhuiz** [comment]
rebased

**Comment by dims** [comment]
@matchstick @mwielgus : moving this to 1.6, please holler if you disagree.

**Comment by nikhiljindal** [comment]
Assigning to @mwielgus to review.
@k8s-bot ok to test

**Comment by quinton-hoole** [comment]
I can't help thinking that these should all be separate flags?

Do we have a precedent one way or the other in kubernetes?  I would have thought that compound, comma-separated command-line parameters like this are not the norm?

**Comment by ethernetdan** [comment]
Removing issue from 1.6 milestone since code freeze has begun:
* If this is a bug fix, please label as `kind/bug` and reapply the milestone.
* If this is a feature for 1.6, an [exception](https://github.com/kubernetes/features/blob/master/EXCEPTIONS.md) should be filed for it before 11:00PM PST on 3/2. Note that most exceptions to code freeze are not accepted.
* If this is a feature for 1.7, feel free to add the 1.7 milestone to the PR.

/cc @kubernetes/release-team @kubernetes/kubernetes-release-managers

**Comment by grodrigues3** [comment]
ping @jianhuiz PR needs rebase

File: federation/pkg/federation-controller/replicaset/replicasetcontroller.go [hunk 1]
```diff
@@ -78,6 +80,34 @@ func parseFederationReplicaSetReference(frs *extensionsv1.ReplicaSet) (*fed.Fede
 	return &frsPref, nil
 }
 
+func initReplicaSetProcessDelays(replicaSetProcessDelays string) {
```
Comment (sttts) [inline federation/pkg/federation-controller/replicaset/replicasetcontroller.go]: don't we have validation in place with proper error handling? `glog.Fatal`s are ugly, especially if they can be triggered by a component config (can they?).

**Comment by sttts** [comment]
> I can't help thinking that these should all be separate flags?
> 
> Do we have a precedent one way or the other in kubernetes? I would have thought that compound, comma-separated command-line parameters like this are not the norm?

Agreeing with @quinton-hoole

**Pull request closed without merging**